odoo/tools/date_utils.py

```diff
@@ -62,3 +65,62 @@ def get_fiscal_year(date, day=31, month=12):
         max_day = calendar.monthrange(date_to.year + 1, date_to.month)[1]
         date_to = type(date)(date.year + 1, month, min(day, max_day))
     return date_from, date_to
+
+
+def start_of(self, value, granularity):
```

Comment:
Expose in fields.Date and fields.Datetime and add the 2 other helpers from pendulum: add() and subtract().